frontend/src/components/billing/billing.tsx

```diff
@@ -0,0 +1,93 @@
+import { AppRoute, DataStatus, NotificationMessage } from 'common/enums/enums';
+import { FC, UserGetResponseWithMoneyBalanceDto } from 'common/types/types';
+import { Button, Spinner } from 'components/common/common';
+import {
+  useAppDispatch,
+  useAppSelector,
+  useEffect,
+  useNavigate,
+} from 'hooks/hooks';
+import { Token } from 'react-stripe-checkout';
+import { notification } from 'services/services';
+import { billingActions } from 'store/actions';
+
+import { DEFAULT_REPLENISH_AMOUNTS, REPLENISH_PUBLIC_KEY } from './common';
+import { ReplenishCardsList } from './components/components';
+import styles from './styles.module.scss';
+
+const MINIMAL_AMOUNT_OF_MONEY_TO_WITHDRAW = 1;
+
+const Billing: FC = () => {
+  const { authDataStatus, user, billingDataStatus, userWithMoneyBalance } =
+    useAppSelector(({ auth, billing }) => ({
+      authDataStatus: auth.dataStatus,
+      user: auth.user,
+      billingDataStatus: billing.dataStatus,
+      userWithMoneyBalance: billing.userWithMoneyBalance,
+    }));
+
+  const navigate = useNavigate();
+
+  if (!user) {
+    navigate(AppRoute.ROOT);
+  }
+
+  const dispatch = useAppDispatch();
+
+  useEffect(() => {
+    dispatch(billingActions.getUserWithMoneyBalance());
+  }, [dispatch]);
+
+  const handleWithdraw = (): void => {
+    if (
+      (userWithMoneyBalance as UserGetResponseWithMoneyBalanceDto).userDetails
+        .moneyBalance >= MINIMAL_AMOUNT_OF_MONEY_TO_WITHDRAW
+    ) {
+      dispatch(billingActions.withdraw());
+    } else {
+      notification.info(NotificationMessage.NOT_ENOUGH_FUNDS_TO_WITHDRAW);
+    }
```

Comment:
I'd prefer to have this logic in action

Reply:
I prefer not event initiate action if the data is invalid. Coaches, please, can you give an advice?

Reply:
I agree with @fshabanov. If we prevented this call in ui, then it would be ok to have this check in component, but in this case we have other services calls so this logic should be inside action